sdk/core/azure-core/inc/azure/core/http/curl/curl.hpp

```diff
@@ -526,22 +530,44 @@ namespace Azure { namespace Core { namespace Http {
                                            : this->m_contentLength == this->m_sessionTotalRead;
     }
 
+    /**
+     * @brief This is a function that can be used to set up curl handle options per connection.
+     *
+     */
+    std::function<CURLcode(CURL* curlHandle)> m_advancedCurlSettingsFn;
```

Comment:
Let's settle on the approach for exposing the options capability for the transport adapter, before checking this in (using callbacks or exposing an transport adapter agnostic options in the ctor).

Reply:
+1 for options in constructor.

I really don't like this callback method.